### PR TITLE
INT-1853 Ensure that the syntax highlight of languages be rendered in GH

### PR DESCRIPTION
**Pull request by hbjORbj**
Screenshot 2023-09-13 at 07 58 04

claap: https://app.claap.io/intuita/ensure-that-the-syntax-highlight-of-languages-be-rendered-in-gh-c-BT2DRbCjzO-fKnJNUOwhgtp

### CODE REVIEW

File: intuita-webview/src/CreateIssue/index.tsx [hunk 1]
```diff
@@ -14,6 +14,17 @@ import './tiptap.css';
 
 const lowlight = createLowlight(common);
 
+const convertHTMLCodeBlockToMarkdownString = (htmlSnippet: string) => {
+	return htmlSnippet.replace(
```
Comment (grzpab) [inline intuita-webview/src/CreateIssue/index.tsx:18]: you can remove the block and the return statement

File: intuita-webview/src/CreateIssue/index.tsx [hunk 2]
```diff
@@ -14,6 +14,17 @@ import './tiptap.css';
...
+		/<pre><code(?:\s+class="language-(.*?)")?>(.*?)<\/code><\/pre>/gs,
+		function (_match, language, content) {
```
Comment (grzpab) [inline intuita-webview/src/CreateIssue/index.tsx:20]: this could be an arrow function I think

File: intuita-webview/src/CreateIssue/index.tsx [hunk 3]
```diff
@@ -14,6 +14,17 @@ import './tiptap.css';
 
 const lowlight = createLowlight(common);
 
+const convertHTMLCodeBlockToMarkdownString = (htmlSnippet: string) => {
```
Comment (grzpab) [inline intuita-webview/src/CreateIssue/index.tsx:17]: why was that needed? aren't we controlling what HTML gets created?

File: intuita-webview/src/CreateIssue/index.tsx [hunk 4]
```diff
@@ -78,8 +88,11 @@ const CreateIssue = (props: Props) => {
 	}, [props.title]);
 
 	useEffect(() => {
+		if (props.loading) {
+			return;
+		}
 		editor?.commands.setContent(props.body, false);
-	}, [editor, props.body]);
+	}, [editor, props.body, props.loading]);
```
Comment (grzpab) [inline intuita-webview/src/CreateIssue/index.tsx:95]: is this a bug fix?

File: intuita-webview/src/CreateIssue/index.tsx (hunk 2 above)
Comment (hbjORbj) [reply intuita-webview/src/CreateIssue/index.tsx:20]: addressed

File: intuita-webview/src/CreateIssue/index.tsx (hunk 1 above)
Comment (hbjORbj) [reply intuita-webview/src/CreateIssue/index.tsx:18]: addressed

File: intuita-webview/src/CreateIssue/index.tsx (hunk 4 above)
Comment (hbjORbj) [reply intuita-webview/src/CreateIssue/index.tsx:95]: yes. this is needed to do the following check
```
if (props.loading) { return; }
```

which is needed to prevent the content changing when user presses "create issue" button. the content will change if `convertHTMLCodeBlockToMarkdownString` results in any change.

File: intuita-webview/src/CreateIssue/index.tsx (hunk 3 above)
Comment (hbjORbj) [reply intuita-webview/src/CreateIssue/index.tsx:17]: The initial value we pass to tiptap editor is a HTML string; otherwise, we can't apply CSS styling.
GH doesn't handle `<pre><code class='language-typescript'>...</code></pre>`. Hence, we run this function before passing it to GH.
